packages/server/graphql/mutations/helpers/bootstrapNewUser.ts

```diff
@@ -76,10 +77,12 @@ const bootstrapNewUser = async (
   }
 
   const isVerified = identities.some((identity) => identity.isEmailVerified)
+  const hasSAMLURL = !!(await getSAMLURLFromEmail(email, dataLoader, false))
+  const isQualifiedForAutoJoin = (isVerified || hasSAMLURL) && isCompanyDomain
```

Comment:
-1 I'm fine with the change as is, but only as a temporary workaround. The real solution would be to add an identity when calling `loginWithSAML`. If we have this workaround and the correct identity adding, we can remove the workaround in a month or so

Reply:
I'm fine splitting it into 3 parts: this workaround, identity fix, removing the workaround. This should be properly reflected in the issues.

Reply:
We call `bootstrapNewUser` from `loginSAML`. Alternative solution is to pass some additional param here:
```
  const authToken = await bootstrapNewUser(tempUser, !isInvited, dataLoader)
  ```
  
  or check if 
  
  `tier: 'enterprise'` which by default for saml
  
  What do you think?

Reply:
Or check `const userId = `sso|${generateUID()}``